tests/unit/test_balancer.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
```

Comment:
**thought**: Maybe we should add a submodule for `scenario`-based unit tests

Reply:
Yeah I think long-term, we should pick one or the other. I used `Harness` here only because we weren't caring **that** much about state in these tests, and we weren't executing events, moreso just calling functions from within the charm context.